engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java

```diff
@@ -870,6 +873,21 @@ private void execute() throws YarnException, IOException
       }
 
       numRequestedContainers += containerRequests.size() - removedContainerRequests.size();
+
+      for (String containerId : dnmgr.containerStopRequests.values()) {
+        Long pendingSince = pendingContainersToKill.get(containerId);
+        if (pendingSince == null) {
+          pendingContainersToKill.put(containerId, System.currentTimeMillis());
+        }
+      }
+
+      for (Map.Entry<String, Long> pendingToKill : pendingContainersToKill.entrySet()) {
+        if (System.currentTimeMillis() - pendingToKill.getValue() > 30 * 1000) {
```

Comment:
Instead of hardcoding 30 * 1000 can we make it configurable or at least define a const variable?